### PR TITLE
Build: Separate type and value imports, enforce with ESLint

### PR DESCRIPTION
**Pull request by IanVS**
Issue: N/A, just something that's been bugging me.  

## What I did

I added the `@typescript-eslint/consistent-type-imports` [rule](https://typescript-eslint.io/rules/consistent-type-imports/), which splits apart type and value imports, and ensures that type imports use the `type` notation.  

This makes it easier to understand at a glance whether a variable is a type or a value by looking at its import.  It also makes it easier for bundlers to strip out the typescript types and help prevent ambiguous imports.

This also helps pave the way for adding a prettier import-sorting plugin, as suggested by @ndelangen on [twitter](https://twitter.com/NorbertdeLangen/status/1585642978856476672).  We'll be able to sort types separately from value imports, or combine them into single statements.

I had to disable `type` imports for Angular, which seemed to cause it to break.  Rather than take them on a case-by-case basis, which could be error prone, I set up eslint for angular to never split out `type` imports.

## How to test

This is mostly a matter of preference.  As long as CI passes it'll be fine to merge, if the team agrees it's worthwhile.

### CODE REVIEW

File: code/addons/a11y/src/components/A11YPanel.tsx [hunk 1]
```diff
@@ -4,16 +4,16 @@ import { styled } from '@storybook/theming';
 
 import { ActionBar, Icons, ScrollArea } from '@storybook/components';
 
-import { AxeResults } from 'axe-core';
+import type { AxeResults } from 'axe-core';
 import { useChannel, useParameter, useStorybookState } from '@storybook/api';
-// eslint-disable-next-line import/no-cycle
+
 import { Report } from './Report';
-// eslint-disable-next-line import/no-cycle
```
Comment (IanVS) [inline code/addons/a11y/src/components/A11YPanel.tsx:11]: I'm not sure why these comments were removed when running eslint with `--fix`, but only three components began erroring, and I added the ignore statements back for those.

**Review by JReinhold — APPROVED** [review]
I don't have strong opinions about this, but your arguments seems solid, and as long as they can be autofixed I'm down.

👍 from me.

**Comment by IanVS** [comment]
Yeah I definitely didn't do all this by hand.  😅 

I'll need to look at the test failures, apparently though.  :(

**Comment by tmeasday** [comment]
Love it @IanVS! Does this solve the problem I was complaining to @ndelangen about yesterday where VSCode auto-inserts non-type imports? Does the following now happen:

1. I auto-complete a type
2. VSCode inserts `import { Type } from '...'` (grr)
3. I save the file
4. Eslint fixes it to `import type { Type } from '...'`?
5.

**Comment by IanVS** [comment]
It should do that so long as ESLint is set up to fix on save in your editor, yes. (I'm having problems with ESLint in VScode in Storybook right now...)

**Review by ndelangen — APPROVED** [review]

**Pull request merged**